portality/migrate/3002_remove_trailing_whitespaces/operations.py

```diff
@@ -0,0 +1,15 @@
+from portality.models import Article
+import csv
+
+
+def remove_trailing_whitespaces_rec(d):
+    for key, value in d.items():
+        if isinstance(value, dict):
+            return remove_trailing_whitespaces_rec(value)
+        if isinstance(value, str):
+            d[key] = value.strip()
+
+
+def remove_trailing_whitespaces(model):
+    remove_trailing_whitespaces_rec(model.__dict__)
```

Comment:
Using `__dict__` isn't right here, that could have some unusual effects.

The better and faster way to do this would be to set `init_with_model:false` in `migrate.json` so that the function just gets given the raw dict of the record in the first case.  Then you can get rid of the "unwrapper" function on line 13.

You also need to `return` the dict from the function, as that's what the upgrade script is expecting.